### PR TITLE
set `allow_superuser` to false as default

### PR DESCRIPTION
**Pull request by kaisecheng**
## Release notes

Running as superuser is disabled by default.  To continue running Logstash as superuser temporarily after the upgrade, set 'allow_superuser' to true.

## What does this PR do?

- set allow_superuser as false by default for v9
- change the buildkite image of ruby unit test to non-root

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

- relates: https://github.com/elastic/logstash/issues/16355
-

### CODE REVIEW

File: logstash-core/lib/logstash/runner.rb [hunk 1]
```diff
@@ -480,7 +480,9 @@ def execute
   def running_as_superuser
     if Process.euid() == 0
       if setting("allow_superuser")
-        deprecation_logger.deprecated("NOTICE: Running Logstash as superuser is not recommended and won't be allowed in the future. Set 'allow_superuser' to 'false' to avoid startup errors in future releases.")
+        deprecation_logger.deprecated("NOTICE: Running Logstash as superuser will be completely disallowed in future versions. " +
+                                        "To prepare for this and avoid startup errors in future releases, it is strongly recommended to set 'allow_superuser' to false now " +
+                                        "and adjust your configuration accordingly.")
```
Comment (kaisecheng) [inline logstash-core/lib/logstash/runner.rb:485]: copy from the companion [pr](https://github.com/elastic/logstash/pull/16555)

Comment (jsvd) [inline logstash-core/lib/logstash/runner.rb:485]:
```suggestion
        logger.warn("NOTICE: Running Logstash as superuser is heavily discourage as it poses a security risk. " +
                                        "It is strongly recommended to set 'allow_superuser' to false.")
```

**Review by jsvd — APPROVED** [review]
LGTM

**Comment by TheVastyDeep** [comment]
@jsvd @kaisecheng You need to update the comment in logstash.yml that says the default is true.

**Pull request merged**